btree/btree.c

```diff
@@ -4237,37 +4134,38 @@ int64_t btree_destroy_tree_tick(struct m0_sm_op *smop)
 
 	M0_PRE(bop->bo_op.o_sm.sm_state == P_INIT);
 	M0_PRE(bop->bo_arbor != NULL);
-	M0_PRE(bop->bo_arbor->t_desc != NULL);
-	M0_PRE(node_invariant(bop->bo_arbor->t_desc->t_root));
-
-	/** The following pre-condition is currently a
-	 *  compulsion as the delete routine has not been
-	 *  implemented yet.
-	 *  Once it is implemented, this pre-condition can be
-	 *  modified to compulsorily remove the records and get
-	 *  the node count to 0.
-	 */
-	M0_PRE(node_count(bop->bo_arbor->t_desc->t_root) == 0);
 
-	tree_put(bop->bo_arbor->t_desc);
-	/**
-	 * TODO: Currently just deleting the tree root node, as the assumption
-	 * is tree will only have root node at this point.
-	 */
-	m0_rwlock_write_lock(&list_lock);
-	ndlist_tlink_del_fini(bop->bo_arbor->t_desc->t_root);
-	m0_rwlock_write_unlock(&list_lock);
+	if (bop->bo_arbor->t_desc != NULL &&  bop->bo_arbor->t_desc->t_ref > 0) {
```

Comment:
Extra space after &&.

Reply:
It has been updated in the latest commit.